www/map-view-controller.js

```diff
@@ -61,30 +63,39 @@ class MapViewControllerImpl {
           // Do nothing.
         },
         () => {
-          console.error("Error at updateNavigationWithLocation")
+          console.error("Error at updateNavigationWithLocation");
         }
       );
     }
   }
 
   _handleOnLocationStatus(payload) {
-    this._sendMessageToViewer("location.update_status", {status: payload.statusName});
+    this._sendMessageToViewer("location.update_status", {
+      status: payload.statusName,
+    });
   }
 
   // ==================================================
-  // MAP-VIEWER MESSAGES: 
+  // MAP-VIEWER MESSAGES:
   // ==================================================
 
   _handleMapViewMessages(m) {
     switch (m.type) {
       case "app.map_is_ready":
-        if (this._onLoadCallback && typeof this._onLoadCallback === 'function') {
+        if (
+          this._onLoadCallback &&
+          typeof this._onLoadCallback === "function"
+        ) {
           this._onLoadCallback(this);
-          console.debug('Map is ready!');
+          console.debug("Map is ready!");
         }
         break;
       case "cartography.poi_selected":
         console.debug(`poi (${m.payload.identifier}) was selected`);
+        this._onPoiSelectedCallback(m.payload);
```

Comment:
Por ahora pode valer, pero imos ter que pasar máis información nesta callback si queremos igualalo ao plugin de flutter (pasamos o poi completo).

Reply:
Como sabe o integrador o formato da función callback e do obxecto que recibe?
En flutter é un pouco diferente ollo, a función recibe un resultado da selección que contén máis cousas.